src/approaches/nsrt_rl_approach.py

```diff
@@ -66,16 +84,121 @@ def learn_from_interaction_results(
         self._online_learning_cycle += 1
         # We get one result per training task.
         for i, result in enumerate(results):
-            states = result.states
-            actions = result.actions
-            traj = LowLevelTrajectory(states,
-                                      actions,
+            traj = LowLevelTrajectory(result.states,
+                                      result.actions,
                                       _is_demo=False,
                                       _train_task_idx=i)
-            self._online_trajectories.append(traj)
+            self._train_task_to_online_traj[i] = traj
 
-        # Replace this with an _RLOptionLearner.
-        self._learn_nsrts(self._initial_trajectories + \
-            self._online_trajectories,
-            online_learning_cycle=self._online_learning_cycle
-        )
+        # This maps each unique _Option to its experience data from the previous
+        # online learning cycle. The experience data is a list of list of tuples
+        # , where the inner list contains experience from one execution of the
+        # option. The tuple is a standard RL experience tuple of (state, action
+        # reward, next_state), with one additional element: the input vector
+        # that the option's regressor received for that state, which is the
+        # contacenation of the state vector for the option's objects and the
+        # relative parameter that the sampler provided. This input vector is
+        # necessary during learning to update the option's regressor.
+        option_to_data: Dict[ParameterizedOption,
+                             List[List[Tuple[State, Array, Action, int,
+                                             State]]]] = {}
+
+        # For each training task, compute the experience data for each _Option
+        # we see used in that training task.
+        for i in range(len(self._train_tasks)):
```

Comment:
let's make a helper function that takes in an `InteractionResult` and produces `option_to_data` for that result; then, join the outputs together in this loop.

Reply:
as part of that refactor, it may as be nice to introduce a new type for the experience tuple (see the bottom of structs.py)